src/authorize.js

```diff
@@ -11,7 +11,27 @@
     if (hash.indexOf('=') === -1) { return; }
     return hash.split('&').reduce(function (m, kvs) {
       var kv = kvs.split('=');
-      m[decodeURIComponent(kv[0])] = decodeURIComponent(kv[1]);
+
+      if (kv[0] === 'state' && kv[1].match(/rsDiscovery/)) {
+        // extract rsDiscovery data from the state param
+        var stateValue = decodeURIComponent(kv[1]);
+        var encodedData = stateValue.substr(stateValue.indexOf('rsDiscovery='))
+                                    .split('&')[0]
+                                    .split('=')[1];
+
+        m['rsDiscovery'] = JSON.parse(atob(encodedData));
+
+        // remove rsDiscovery param
+        stateValue = stateValue.slice(stateValue.indexOf('rsDiscovery='),
+                                      encodedData.lenth + 12);
```

Comment:
`encodedData.lenth` should be `encodedData.length` (typo). It looks like this would remove all state data if there were keys after `rsDiscovery`